src/lib/remark-plugins/remark-plugin-remove-h1/index.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import { Heading } from 'mdast'
+import { Plugin, Transformer } from 'unified'
+import { visit, SKIP, Parent, Node } from 'unist-util-visit'
+
+/**
+ * This plugin removes any `<h1 />` element from the `remark` AST
+ * no which it is run.
+ */
+const remarkPluginRemoveH1: Plugin = (): Transformer => {
```

Comment:
We now render a `DocsPageHeading` component with an `h1` outside the MDX content area for nearly all docs pages (likely short-lived exception: Packer plugins).

We run this plugin to avoid rendering a duplicative `h1` within the content area.

Reply:
Do we have a content conformance check in the works for this?

Reply:
💭  I worry if folks use an h1 and then get no authoring feedback about it getting removed.

Reply:
I believe there is a check in place to ensure all documents have an `h1` in their content.

Re: removal of `h1`, I think it's more accurate to say "we render it outside the MDX content area, rather than within it" (we happen to use remark plugins to make that happen, the `anchor-links` plugin extracts all headings including the `h1`, the `remove-h1` plugin ensures we don't render the `h1` in both areas). So we're still rendering the `h1` that authors write in MDX, just in a different spot. With that in mind, my take was we don't need to warn authors about it, since it's more an implementation detail than anything else.

Reply:
Okay sweet, after looking over the rest of the code I see now how the `h1`s were being used within the document and forgot that docs pages don't have a proper `title`, so this makes sense!